experimental/uefi/loader/src/crosvm.rs

```diff
@@ -14,32 +14,88 @@
 // limitations under the License.
 //
 
-use crate::vmm::{Params, Vmm};
+use crate::{
+    vmm::{Params, Vmm},
+    ReadWrite,
+};
 use anyhow::Result;
 use async_trait::async_trait;
 use command_fds::tokio::CommandFdAsyncExt;
 use log::info;
 use std::{
+    io::{Error, ErrorKind, Read, Write},
     net::Shutdown,
-    os::unix::{
-        io::{AsRawFd, OwnedFd},
-        net::UnixStream,
-    },
+    os::unix::{io::AsRawFd, net::UnixStream},
     process::Stdio,
 };
+use vsock::VsockStream;
+
+/// The guest VM context ID for virtio vsock connecitons.
+const VSOCK_GUEST_CID: u32 = 3;
+
+/// The guest VM virtio vsock port.
+const VSOCK_GUEST_PORT: u32 = 1024;
 
 pub struct Crosvm {
     console: UnixStream,
     instance: tokio::process::Child,
 }
 
+/// A wrapper for a vsock stream that can be used as a communication channel between the loader and
+/// the guest VM.
+///
+/// The vsock channel can only be created after the VM is booted, so we start without a stream and
+/// only connect later when we try to use it for the first time.
+pub struct VsockStreamChannel {
+    stream: Option<VsockStream>,
+}
+
+impl VsockStreamChannel {
+    /// Ensures that the inner stream is connected to the VM.
```

Comment:
I think here we do more than `ensure_stream`---we also connect the stream if it is not there, no? Maybe we can update the comment to reflect this, i.e., something along the lines "ensures the inner stream is connected ... by connecting/creating the stream if necessary" ?

Reply:
Done.